packages/astro/src/content/template/types.d.ts

```diff
@@ -13,8 +13,27 @@ declare module 'astro:content' {
 	export type CollectionEntry<C extends keyof typeof entryMap> =
 		(typeof entryMap)[C][keyof (typeof entryMap)[C]];
 
+	// TODO: Remove this when having this fallback is no longer relevant. 2.3? 3.0? - erika, 2023-04-04
+	/**
+	 * @deprecated
```

Comment:
Should the existing behavior here be changing if it's being marked as deprecated? Or is it just here for the visual cue in editors

Reply:
Just there for the visual cue. Since the feature is experimental, we don't care much about API stability and maintaining both options doesn't really work anyway.